better error when user is not logged in

it currently reports 
'error in requirement' 

cc @aeckard87

Actually I get this when not logged in..
```
10:34 $ cf usage-report
panic: interface conversion: interface {} is nil, not float64

goroutine 1 [running]:
github.com/krujos/usagereport-plugin/apihelper.(*APIHelper).GetOrgs(0xc420112200, 0x410212, 0xc420112230, 0x10, 0x10, 0x70cb00)
        /home/drennalls/.gvm/pkgsets/go1.8.3/global/src/github.com/krujos/usagereport-plugin/apihelper/apihelper.go:63 +0x7c4
main.(*UsageReportCmd).getOrgs(0xc4201121b0, 0x1, 0x1, 0x0, 0x0, 0x787e1e)
        /home/drennalls/.gvm/pkgsets/go1.8.3/global/src/github.com/krujos/usagereport-plugin/usagereport.go:100 +0x4b
main.(*UsageReportCmd).UsageReportCommand(0xc4201121b0, 0xc420010530, 0x1, 0x1)
        /home/drennalls/.gvm/pkgsets/go1.8.3/global/src/github.com/krujos/usagereport-plugin/usagereport.go:83 +0x3d1
main.(*UsageReportCmd).Run(0xc4201121b0, 0x90a120, 0xc4201121c0, 0xc420010530, 0x1, 0x1)
        /home/drennalls/.gvm/pkgsets/go1.8.3/global/src/github.com/krujos/usagereport-plugin/usagereport.go:189 +0xfa
github.com/cloudfoundry/cli/plugin.Start(0x904380, 0xc4201121b0)
        /home/drennalls/.gvm/pkgsets/go1.8.3/global/src/github.com/cloudfoundry/cli/plugin/plugin_shim.go:34 +0x28d
main.main()
        /home/drennalls/.gvm/pkgsets/go1.8.3/global/src/github.com/krujos/usagereport-plugin/usagereport.go:194 +0x47
```
Took me a while to realize not being logged in was the reason for the error though :)

--- a/apihelper/apihelper.go
+++ b/apihelper/apihelper.go
@@ -54,9 +54,20 @@ func New(cli plugin.CliConnection) CFAPIHelper {
 	return &APIHelper{cli}
 }
 
+func cfCurl(cli plugin.CliConnection, path string) (map[string]interface{}, error) {
+	orgsJSON, err := cfcurl.Curl(cli, path)
+
+	if orgsJSON["error_code"] != nil {
+		cfcurlErr := errors.New("Error calling CF API: " + orgsJSON["description"].(string))
+		return nil, cfcurlErr
+	}
+
+	return orgsJSON, err
+}
+
 //GetOrgs returns a struct that represents critical fields in the JSON
 func (api *APIHelper) GetOrgs() ([]Organization, error) {
-	orgsJSON, err := cfcurl.Curl(api.cli, "/v2/organizations")
+	orgsJSON, err := cfCurl(api.cli, "/v2/organizations")
 	if nil != err {
 		return nil, err
 	}
@@ -64,7 +75,7 @@ func (api *APIHelper) GetOrgs() ([]Organization, error) {
 	orgs := []Organization{}
 	for i := 1; i <= pages; i++ {
 		if 1 != i {
-			orgsJSON, err = cfcurl.Curl(api.cli, "/v2/organizations?page="+strconv.Itoa(i))
+			orgsJSON, err = cfCurl(api.cli, "/v2/organizations?page="+strconv.Itoa(i))
 		}
 		for _, o := range orgsJSON["resources"].([]interface{}) {
 			theOrg := o.(map[string]interface{})
@@ -86,7 +97,7 @@ func (api *APIHelper) GetOrgs() ([]Organization, error) {
 func (api *APIHelper) GetOrg(name string) (Organization, error) {
 	query := fmt.Sprintf("name:%s", name)
 	path := fmt.Sprintf("/v2/organizations?q=%s&inline-relations-depth=1", url.QueryEscape(query))
-	orgsJSON, err := cfcurl.Curl(api.cli, path)
+	orgsJSON, err := cfCurl(api.cli, path)
 	if nil != err {
 		return Organization{}, err
 	}
@@ -116,7 +127,7 @@ func (api *APIHelper) orgResourceToOrg(o interface{}) Organization {
 
 //GetQuotaMemoryLimit retruns the amount of memory (in MB) that the org is allowed
 func (api *APIHelper) GetQuotaMemoryLimit(quotaURL string) (float64, error) {
-	quotaJSON, err := cfcurl.Curl(api.cli, quotaURL)
+	quotaJSON, err := cfCurl(api.cli, quotaURL)
 	if nil != err {
 		return 0, err
 	}
@@ -125,7 +136,7 @@ func (api *APIHelper) GetQuotaMemoryLimit(quotaURL string) (float64, error) {
 
 //GetOrgMemoryUsage returns the amount of memory (in MB) that the org is consuming
 func (api *APIHelper) GetOrgMemoryUsage(org Organization) (float64, error) {
-	usageJSON, err := cfcurl.Curl(api.cli, org.URL+"/memory_usage")
+	usageJSON, err := cfCurl(api.cli, org.URL+"/memory_usage")
 	if nil != err {
 		return 0, err
 	}
@@ -137,7 +148,7 @@ func (api *APIHelper) GetOrgSpaces(spacesURL string) ([]Space, error) {
 	nextURL := spacesURL
 	spaces := []Space{}
 	for nextURL != "" {
-		spacesJSON, err := cfcurl.Curl(api.cli, nextURL)
+		spacesJSON, err := cfCurl(api.cli, nextURL)
 		if nil != err {
 			return nil, err
 		}
@@ -164,7 +175,7 @@ func (api *APIHelper) GetSpaceApps(appsURL string) ([]App, error) {
 	nextURL := appsURL
 	apps := []App{}
 	for nextURL != "" {
-		appsJSON, err := cfcurl.Curl(api.cli, nextURL)
+		appsJSON, err := cfCurl(api.cli, nextURL)
 		if nil != err {
 			return nil, err
 		}

--- a/apihelper/apihelper_test.go
+++ b/apihelper/apihelper_test.go
@@ -186,4 +186,18 @@ var _ = Describe("UsageReport", func() {
 	})
 
 	// TODO need tests for no spaces and no apps in org.
+
+	Describe("error calling CF API", func() {
+		var errorJSON []string
+
+		BeforeEach(func() {
+			errorJSON = slurp("test-data/not-authenticated.json")
+		})
+
+		It("should return an error when CF API call fails", func() {
+			fakeCliConnection.CliCommandWithoutTerminalOutputReturns(errorJSON, nil)
+			_, err := api.GetOrgs()
+			Expect(err.Error()).To(Equal("Error calling CF API: Authentication error"))
+		})
+	})
 })

--- a/apihelper/test-data/not-authenticated.json
+++ b/apihelper/test-data/not-authenticated.json
@@ -1,0 +1,5 @@
+{
+   "description": "Authentication error",
+   "error_code": "CF-NotAuthenticated",
+   "code": 10002
+}

